vsts/windowsLinuxAndAndroidBuildMatrixConfig.yaml

```diff
@@ -1,299 +1,248 @@
-      TestGroup34:
-        ANDROID_TEST_GROUP_ID: TestGroup34
-      TestGroup35:
-        ANDROID_TEST_GROUP_ID: TestGroup35
-      TestGroup36:
-        ANDROID_TEST_GROUP_ID: TestGroup36
-      TestGroup37:
-        ANDROID_TEST_GROUP_ID: TestGroup37
-      TestGroup38:
-        ANDROID_TEST_GROUP_ID: TestGroup38
-      TestGroup39:
-        ANDROID_TEST_GROUP_ID: TestGroup39
-      
-  displayName: Android Test
-  dependsOn: AndroidBuild
-  steps:
-  - task: PowerShell@2
-    displayName: 'determine if testing needed'
-    condition: always()
-    inputs:
-      targetType: 'filePath'
-      filePath: ./vsts/determine_if_android_test_group_needs_to_run.ps1
-    env:
-      TEST_GROUP_ID: $(ANDROID_TEST_GROUP_ID)
-      IS_BASIC_TIER_HUB: $(IS-BASIC-TIER-HUB)
-      TARGET_BRANCH: $(System.PullRequest.TargetBranch)
+      - task: CopyFiles@2
+        displayName: 'Copy Test Results to Artifact Staging Directory'
+        inputs:
+          SourceFolder: '$(Build.SourcesDirectory)'
+          Contents: |
+            **/*.trx
+            **/*.xml
+          TargetFolder: '$(Build.ArtifactStagingDirectory)'
+        continueOnError: true
+        condition: always()
 
-  - task: PowerShell@2
-    displayName: 'Install app center cli'
-    condition: eq(variables['task.android.needToRunTestGroup'], 'yes')
-    inputs:
-      targetType: 'filePath'
-      filePath: './vsts/install_appcenter_cli.ps1'
+      - task: PublishBuildArtifacts@1
+        displayName: 'Publish Artifact Staging Directory'
+        continueOnError: true
+        condition: always()
 
-  - task: DownloadPipelineArtifact@0
-    condition: eq(variables['task.android.needToRunTestGroup'], 'yes')
-    inputs:
-      artifactName: 'androidBuildFiles'
-      targetPath: $(Build.ArtifactStagingDirectory)/iot-e2e-tests/android/app/build/outputs/apk
-  
-  - task: AppCenterTest@1
-    condition: eq(variables['task.android.needToRunTestGroup'], 'yes')
-    displayName: 'E2E with Visual Studio App Center'
-    inputs:
-      appFile: '$(Build.ArtifactStagingDirectory)/iot-e2e-tests/android/app/build/outputs/apk/debug/app-debug.apk'
-      frameworkOption: espresso
-      espressoBuildDirectory: '$(Build.ArtifactStagingDirectory)/iot-e2e-tests/android/app/build/outputs/apk/androidTest/debug'
-      serverEndpoint: 'AppCenter connection aziotclb'
-      appSlug: 'Azure-Iot-Sdk/androide2e'
-      devices: 'Azure-Iot-Sdk/api28_2'
-      runOptions: '--test-parameter annotation=com.microsoft.azure.sdk.iot.android.helper.$(ANDROID_TEST_GROUP_ID)'
-      showDebugOutput: true
\ No newline at end of file
+      - task: PublishTestResults@2
+        displayName: 'Publish Test Results'
+        inputs:
+          mergeTestResults: true
+        continueOnError: true
+        condition: always()
+
+  ### Android, Multi configuration build (12 different test groups to cover) ###
+  - job: AndroidBuild
+    timeoutInMinutes: 45
+    pool:
+      name: Hosted VS2017
```

Comment:
The [doc](https://docs.microsoft.com/en-us/azure/devops/pipelines/ecosystems/android?view=azure-devops#test-on-the-android-emulator) mentioned that: The Android Emulator is currently available only on the Hosted macOS agent.

But we seem to be using a win image here?

Reply:
Ah, I see the AndroidTest uses macOS-latest

Reply:
We build the android APKs on a windows image, but the Android Test portion of this yaml is all on a Mac OS